fix: Removed an extraneous period, and added more explanations of Wakers

ckaran opened the pull request:
I removed a spurious period (first commit), and added a longer explanation of Wakers and what they're for.  Might be longer than necessary, but I thought it might be necessary as the current explanation of how Wakers are created could imply that each Executor makes exactly one Waker, rather than one per registered Future.

cfsamson commented:
Thanks. This is great! 

I really liked the example with the cancellable future! 

One question, have you tried that way of implementing a cancellable future this way or seen any implementations like the one you outlined? (I can't see why it wouldn't work though).

cfsamson commented:
I moved the additional notes to the [bonus segment of chapter 2](https://cfsamson.github.io/books-futures-explained/1_futures_in_rust.html#bonus-section---additional-notes-on-futures-and-wakers) since I felt it belonged there. 

I also added a thanks in the [credit section](https://cfsamson.github.io/books-futures-explained/#credits-and-thanks) for the contribution. I hope that's ok. Let me know if it's not and I'll remove it.

ckaran commented:
> One question, have you tried that way of implementing a cancellable future this way or seen any implementations like the one you outlined? (I can't see why it wouldn't work though).

I haven't, but I think it's similar to what you get from [`abortable()`](https://docs.rs/futures/0.3.8/futures/future/fn.abortable.html). **NOTE!** I haven't really dug through that code, I skimmed it quite a while ago, and may have bits of it drifting through my subconscious... or I might have invented it myself.  I really don't know... :man_shrugging: 

> I moved the additional notes to the bonus segment of chapter 2 since I felt it belonged there.

Works for me, thank you.

> I also added a thanks in the credit section for the contribution. I hope that's ok. Let me know if it's not and I'll remove it.

Thank you, I appreciate it! :grinning:

cfsamson commented:
>I haven't, but I think it's similar to what you get from abortable(). NOTE! I haven't really dug through that code, I skimmed it quite a while ago, and may have bits of it drifting through my subconscious... or I might have invented it myself. I really don't know... 🤷‍♂️

Sounds good. I've looked over it, and as far as I can tell it builds on the same idea you explained in the paragraph about cancellable futures.
